Control/Parallel/Strategies.hs

```diff
@@ -359,19 +359,30 @@ withStrategyIO :: Strategy a -> a -> IO a
 withStrategyIO = flip usingIO
 
 -- | Compose two strategies sequentially.
--- This is the analogue to function composition on strategies.
 --
--- For any strategies @strat1@, @strat2@, and @strat3@,
+-- > strat2 `dot` strat1 == strat2 . withStrategy strat1
+--
+-- 'dot' is associative:
 --
 -- > (strat1 `dot` strat2) `dot` strat3 == strat1 `dot` (strat2 `dot` strat3)
--- > strat1 `dot` strat1 = strat1
--- > strat1 `dot` r0 == strat1
 --
--- > strat2 `dot` strat1 == strat2 . withStrategy strat1
+-- 'r0' and 'rseq' are one-sided identities of 'dot':
+--
+-- > strat `dot` r0 == strat
+-- > strat `dot` rseq == strat
 --
+-- Furthermore, since strategies should only force and spark computations
+-- (that is, they don't actually change any values),
+--
+-- > strat `dot` strat == strat
+--
+-- Perhaps unexpectedly, @r0 `dot` strat == r0@.
```

Comment:
No, it isn't. `r0` will never apply `strat`, but the left hand side will apply `strat`, if and when the value is evaluated to WHNF.

Reply:
Oh, right, that's even more confusing. `strat` isn't used when applying `runEval` (or `using`), but when evaluating the result of that.